### DISCUSSION TITLE
training time length

### DISCUSSION THREAD

**Discussion by mohsenhrt** [hamedhemati/Tacotron-2-Persian#10] [General]
Hi and thank you for your codes. How long does it take?(hours)

**Comment by HamedHemati** [comment]
Hi!
It depends on the batch size you want to use. With a batch size of 64, it shouldn't take more than 1.5 days to converge. Also, the mel-spec computation is done online during training, so if you have CPU computation limits, you might want to consider first pre-computing the mel-spec features and then training the model. The vocoder is not trained on the target speaker's voice; I just used a pre-trained "universal" vocoder.
Nevertheless, the dataset is not a high-quality dataset for TTS and the high variation in the recording conditions and background noise poses instability during training. It's probably better to give it a try on another dataset, maybe the one below (haven't tried it myself) or a newer dataset.
https://www.kaggle.com/moradi/persian-texttospeech-audio

**Comment by mohsenhrt** [comment]
Thank you,
How long does it take to get result from test? When I read your README.md I see you  "...hard-coded in the` commonvoice_fa` preprocessor" where is your hard - code?

with best regards

**Comment by HamedHemati** [comment]
The mel-spec generation part is relatively fast; it takes around 1-2 seconds on a CPU depending on the length of the input text. The vocoder (WaveRNN) part however is slow. Sometimes it takes up to ~10-15 seconds for long audios. 
The code is not really efficient for production. WaveRNN could be replaced by a faster vocoder like Parallel WaveGAN or HiFiGAN. Check out this repository (if you haven't) for some cool recent models:
https://github.com/coqui-ai/TTS

By hard coding I meant that the speaker ID of the target speaker in the Mozilla CommonVoice dataset is directly inserted into the code in the `tac2persian/data_preprocessing/preprocess_commonvoice_fa.py` file line 108 where `target_speakers = ["0d358...`.
